### PR TITLE
Add GCE Request Count Metric

### PR DESCRIPTION
**Pull request by swetharepakula**
Add GCE Request Count Metric that measures the number of requests the NEG Controller sends to the NEG API

/assign @gauravkghildiyal

### CODE REVIEW

**Comment by swetharepakula** [comment]
This is based on #2153 . I will rebase and remove hold once that is merged

**Comment by swetharepakula** [comment]
https://github.com/kubernetes/ingress-gce/pull/2153 has merged, so removing hold
/hold cancel

File: pkg/neg/metrics/metrics.go [hunk 1]
```diff
@@ -282,6 +317,13 @@ func PublishAnnotationMetrics(annotationSize int, labelNumber int) {
 	LabelNumber.Observe(float64(labelNumber))
 }
 
+// PublishGCERequestCountMetrics publishes collected metrics for GCE Request Counts
+func PublishGCERequestCountMetrics(start time.Time, requestType string, err error) {
+	result := getResult(err)
+	GCERequestLatency.WithLabelValues(requestType, result).Observe(time.Since(start).Seconds())
+	GCERequestCount.WithLabelValues(requestType, getResult(err)).Inc()
```
Comment (gauravkghildiyal) [inline pkg/neg/metrics/metrics.go:324]: nit: Possible replacement: getResult(err) -> result

Comment (swetharepakula) [reply pkg/neg/metrics/metrics.go]: Yes, thanks for catching that!

**Pull request merged**